src/stores/roles/actions.ts

```diff
@@ -0,0 +1,35 @@
+import { doRequest, get, post, put } from '@/services/http'
+import { getRoles } from './getters'
+import type { Role } from '@/domain/Role'
+import { roles } from './state'
+import type { Knowledge } from '@/domain/Knowledge'
+import type { Answer } from '@/domain/Answer'
+import type { QuizAnswers } from '@/domain/QuizAnswer'
+
+function setRoles(newRoles: Role[]): void {
+  roles.value = newRoles
+}
+
+async function fetchRoles(): Promise<void> {
+  if (getRoles?.value?.length > 0) {
+    return
+  }
+
+  const fetchedRoles = await get<Role[]>('/roles')
+
+  setRoles(fetchedRoles)
+}
+
+async function getQuizResult(quizAnswer: QuizAnswers) {
+  const result = await post('/calculate-score', quizAnswer)
+  return result
+}
+
+const changeQuizAnswers = (knowledges: Knowledge[], quizAnswer: QuizAnswers) => {
```

Comment:
1 - name of the method
2 - 
```suggestion
const changeQuizAnswers = (knowledges: Knowledge[], quizAnswers: QuizAnswers) => {
```